core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -410,11 +410,13 @@ class TeamPolicyInternal<Kokkos::Cuda, Properties...>
                  : functor_value_traits::value_size(f)));
     KOKKOS_ASSERT(block_size > 0);
 
-    // Currently we require Power-of-2 team size for reductions.
-    int p2 = 1;
-    while (p2 <= block_size) p2 *= 2;
-    p2 /= 2;
-    return p2 / impl_vector_length();
+    // Currently we require power-of-two team size for reductions.
```

Comment:
I am not actually sure that this comment is still accurate.

Reply:
We might consider opening another issue on this as this PR fixes the reported error.